File: miner/pow.py
import hashlib, sys, json

originalString = sys.argv[1]
difficulty = int(sys.argv[2])
nonce = 0

stringToHash = originalString + str(nonce)

hashed = hashlib.sha256(stringToHash).hexdigest()
while hashed[:difficulty] != "0" * difficulty:
  nonce += 1
  stringToHash = originalString + str(nonce)
  hashed = hashlib.sha256(stringToHash).hexdigest()

response = nonce

print response
sys.stdout.flush()

File: miner/pow.js
const SHA256 = require("crypto-js/sha256")


let originalString = process.argv[2]
let difficulty = parseInt(process.argv[3])
let nonce = 0

let stringToHash = originalString + nonce

let hash = SHA256(stringToHash).toString()
while(hash.substring(0, difficulty) !== Array(difficulty + 1).join("0")){
  nonce += 1
  stringToHash = originalString + nonce
  hash = SHA256(stringToHash).toString()
}

console.log(hash)




